### ISSUE TITLE
Comment boxes parse their contents...?

### ISSUE THREAD

**Issue by jamshark70** [pure-data/pure-data#1291]
Using the "expected vs actual" bug report style:

- Expected behavior: Comments in programming environments are free text (pure text), exempt from parsing.

- Actual behavior: Pd parses floats in comment boxes.

Specific case: I assigned students in a class to create a patch with specific requirements, working in small groups. They could identify the students using a comment. Student ID numbers at my school are of the format yyyyddxxx: four digits for their enrollment year, two for the department and three for the ordinal, e.g. 201803001. The students naïvely wrote e.g. "Name One 201803001 Name Two 201803002" etc.

Upon opening the patch, I see 2.01803e+008 -- single-precision float format has lost the ordinal digits (which are, of course, the ones I really need).

Sure, I can advise students to write only the ordinal digits next time.

But... `/* this is a comment */` skips parsing in (I would guess) every other programming environment, and I can't think of a good reason for Pd to break that principle.

(Of course it's easy to see why it happens -- the contents of all user input into boxes are parsed, just not verified in the case of comments. That doesn't justify the current behavior.)

**Comment by umlaeute** [comment]
yes. this is a known issue. there seems to be no bug-report yet, so thanks for reporting it.

i'm afraid, priority is quite low for this (given the way the parser works).

i *think* the way forward is to fix the way comments are saved, rather than how they are parsed.

Related: https://github.com/pure-data/pure-data/issues/1083

**Comment by Spacechild1** [comment]
> i think the way forward is to fix the way comments are saved, rather than how they are parsed.

Once a string token has been parsed as a float atom we don't have access to the original string representation resp. the original precision, so I don't think this can be fixed by changing the saving logic.

IMO, Tcl should send the string as is (including all whitespace and newlines), which would practically circumvent the parser. We wouldn't even have to change the saving logic, as the whole text would be contained in a single symbol. However, for drawing the text, the GUI would need to parse the symbol for word wrap and line breaks. The only problem I see is that some people rely on the current behavior of semicolons introducing a line break, but we could just keep treating semicolons specially.

**Comment by porres** [comment]
> (including all whitespace and newlines)

that'd be great!!!

**Comment by umlaeute** [comment]
i just looked into this, and it's actually really simple to fix (if you can live with each comment being stored in a separate symbol; the symbols themselves would obviously be rather long; otoh, when looking all the comments in doc/, this would reduce the number of symbols required from 7760 to 4083).

anyhow, a real fix would also require #824 to be fixed first.

**Comment by jamshark70** [comment]
> i just looked into this, and it's actually really simple to fix (if you can live with each comment being stored in a separate symbol; the symbols themselves would obviously be rather long...)

I'd be OK with that.
